### PR TITLE
Remove OVM v1 reference and update branch name of OVM v2

### LINKED ISSUE #295: Remove OVM 1.0 from the Running a development node guide
In the "Running a Development Node" tutorial, there is reference to an OVM 1.0 docker image. This is no longer relevant and causes confusion. Remove reference to OVM 1.0 in [this page](https://community.optimism.io/docs/developers/l2/dev-node.html). 

Once removed, please follow the tutorial locally to make sure that it is correct and up to date.

### LINKED ISSUE #296: Update the node tutorial with the correct branch
In the ["Running a Development Node"](https://community.optimism.io/docs/developers/l2/dev-node.html) doc the OVM 2.0 clone instructions are outdated. It currently says: 
`git clone https://github.com/ethereum-optimism/optimism.git -b regenesis/0.5`
But instead it should be:
`git clone https://github.com/ethereum-optimism/optimism.git -b develop`

Please update this within the docs.

### PR DESCRIPTION
**Pull request by LeifuChen**
**Description**
Remove OVM 1.0 docker image and update the correct branch of OVM 2.0

**Metadata**
- Fixes #295 
- Fixes #296

### CODE REVIEW

File: src/docs/developers/l2/dev-node.md [hunk 1]
```diff
@@ -57,16 +57,10 @@ If you want to download the images, perform these steps:
 1. Clone the [Optimism monorepo](https://github.com/ethereum-optimism/optimism).
    The branch to clone depends on the OVM version you need.
```
Comment (smartcontracts) [inline src/docs/developers/l2/dev-node.md:58]: I think we should just get rid of this sentence entirely and just use the code snippet for cloning the repository.

**Comment by smartcontracts** [comment]
Looking good! Thanks for tackling this :-)

File: src/docs/developers/l2/dev-node.md (hunk 1 above)
Comment (LeifuChen) [reply src/docs/developers/l2/dev-node.md:58]: Sure. I will make changes.

Comment (LeifuChen) [reply src/docs/developers/l2/dev-node.md]: @smartcontracts The change has been made.

**Review by smartcontracts — APPROVED** [review]

**Comment by smartcontracts** [comment]
Approved, just need one more approval from @Nicca42

**Pull request merged**